tracer/contrib/database/sqltraced/sqltraced.go

```diff
@@ -0,0 +1,458 @@
+package sqltraced
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/dd-trace-go/tracer"
+	"github.com/DataDog/dd-trace-go/tracer/ext"
+)
+
+// Warning: the `name` must be different from the name used by the initial driver.
+// E.g. for the mysql driver you can't use "mysql, but you can use "tracedMysql".
+func Register(name, service string, driver driver.Driver, trc *tracer.Tracer) {
+	if driver == nil {
+		panic("RegisterTracedDriver: driver is nil")
+	}
+	if trc == nil {
+		trc = tracer.NewTracer()
+	}
+
+	trace := Trace{
```

Comment:
`Trace` is a bad name for this info. It's already a concept (a collection of spans) and it is not this. we should just pass in the params.